.github/workflows/main.yml

```diff
@@ -32,67 +32,61 @@ jobs:
           chmod a+x appsign.sh || exit 1
 
       - name: Compile with codesign (DEBUG)
-        if: github.repository_owner == 'acidanthera'
+        if: github.repository_owner == 'acidanthera' && github.event_name != 'pull_request'
         env:
           MAC_CERTIFICATE_PASSWORD: ${{ secrets.MAC_CERTIFICATE_PASSWORD }}
           MAC_ACCOUNT_NAME: ${{ secrets.MAC_ACCOUNT_NAME }}
           MAC_ACCOUNT_PASSWORD: ${{ secrets.MAC_ACCOUNT_PASSWORD }}
         run: DEPLOY_SCRIPT="$(pwd)/appsign.sh" xcodebuild -jobs 1 -configuration Debug
 
       - name: Compile with codesign (RELEASE)
-        if: github.repository_owner == 'acidanthera'
+        if: github.repository_owner == 'acidanthera' && github.event_name != 'pull_request'
         env:
           MAC_CERTIFICATE_PASSWORD: ${{ secrets.MAC_CERTIFICATE_PASSWORD }}
           MAC_ACCOUNT_NAME: ${{ secrets.MAC_ACCOUNT_NAME }}
           MAC_ACCOUNT_PASSWORD: ${{ secrets.MAC_ACCOUNT_PASSWORD }}
         run: DEPLOY_SCRIPT="$(pwd)/appsign.sh" xcodebuild -jobs 1 -configuration Release
 
       - name: Compile (DEBUG)
-        if: github.repository_owner != 'acidanthera'
+        if: github.repository_owner != 'acidanthera' || github.event_name == 'pull_request'
         run: xcodebuild -jobs 1 -configuration Debug
 
       - name: Compile (RELEASE)
-        if: github.repository_owner != 'acidanthera'
+        if: github.repository_owner != 'acidanthera' || github.event_name == 'pull_request'
         run: xcodebuild -jobs 1 -configuration Release
 
       - name: Upload to Artifacts
-        if: github.repository_owner == 'acidanthera'
+        if: github.repository_owner == 'acidanthera' && github.event_name != 'pull_request'
         uses: actions/upload-artifact@v2
         with:
           name: Artifacts
           path: build/*/*.dmg
 
       - name: Upload to Release
-        if: github.event_name == 'release' && github.repository_owner == 'acidanthera'
+        if: github.event_name == 'release' && github.repository_owner == 'acidanthera' && github.event_name != 'pull_request'
```

Comment:
This one is always true, no need to check for `pull_request` here and below where there is a `release` check already.